mw-webapp/src/utils/DateUtils.ts

```diff
@@ -48,21 +49,22 @@ export class DateUtils {
    * Get dates between two dates
    */
   public static getDatesBetween(startDate: Date, endDate: Date): Date[] {
-    const timeDifferenceInMilliseconds = Math.abs(endDate.getTime() - startDate.getTime());
-    const daysAmountInRange = Math.ceil(timeDifferenceInMilliseconds / DAY_MILLISECONDS);
-    const datesList = new Array(daysAmountInRange)
-      .fill(null)
-      .map((item, i) => {
 
-        /**
-         * TODO #573: Correction need because method getShortISODateValue return time in GMT format but we need UTC format
-         */
-        const correction = DAY_MILLISECONDS;
+    const isoStartDate = DateUtils.getShortISODateValue(startDate);
+    const isoEndDate = DateUtils.getShortISODateValue(endDate);
 
-        return new Date(startDate.getTime() + (i * DAY_MILLISECONDS) + correction);
-      });
+    const currentDate = new Date(isoStartDate);
+    const lastDate = new Date(isoEndDate);
 
-    return datesList;
+    lastDate.setDate(lastDate.getDate() + ONE_DAY);
+
+    const daysCount = (lastDate.getTime() - currentDate.getTime()) / DAY_MILLISECONDS;
+
+    const dateList = Array.from(
+      {length: daysCount},
+      (_, i) => new Date(currentDate.getTime() + (i * DAY_MILLISECONDS)));
+
```

Comment:
Too difficult to understand. Is it possible to make it easier?

Reply:
fixed